### ISSUE TITLE
Migrate to CircleCI 2.0

### ISSUE THREAD

**Issue by c24t** [census-instrumentation/opencensus-python#402]
We're seeing a deprecation notice on https://circleci.com/gh/census-instrumentation/opencensus-python, time to upgrade.

![screen shot 2018-11-19 at 5 42 44 pm](https://user-images.githubusercontent.com/43799105/48746084-c56b8000-ec22-11e8-9733-d96d7f18a8bb.png)

**Comment by c24t** [comment]
See https://circleci.com/docs/2.0/workflows/#git-tag-job-execution for notes on updating our build-on-tag workflow.

**Comment by c24t** [comment]
Consider moving to travis, like the other OC client libraries.

**Comment by reyang** [comment]
@lzchen

**Comment by bogdandrutu** [comment]
OpenTelemetry is also going to use Travis.

**Comment by lzchen** [comment]
Travis CI gives us a lot of functionality and freedom for configurations, especially related to running different versions of Python. To take advantage of these customizations, we could consider moving our build automation from nox to tox (https://tox.readthedocs.io/en/latest/). There's a plugin available that integrates tox with TravisCI (https://tox-travis.readthedocs.io/en/stable/)

Nox's ability to run specific versions of Python per function can be replicated using functionality available in TravisCI. Nox also lacks a lot of the configurations that would be useful (such as caching dependencies). There's also the case of nox not being widely used so there's is very little support or documentation on it.

**Comment by lzchen** [comment]
There's also version specific logic that exists which will fail the coverage aggregation check (because nox aggregates results from different run versions). A possibility could be lowering the coverage threshold for aggregated test results.

**Comment by reyang** [comment]
@lzchen, what's your thinking on running test against different versions of python on local dev box?

**Comment by lzchen** [comment]
@reyang  Running tests on different versions locally would be done sequentially, which might take a lot of time, especially with the installation of dependencies. With Travis, new virtual envs would be boot up at the same time and run in parallel and we can configure them to share from the same dependency list.

**Comment by bu3alwa** [comment]
Is this still needed? If so can I give it a go?

**Comment by lzchen** [comment]
Yes! The `tox.ini` file is already setup and we are using Travis CI but have not completed migrations yet.
Take what's left in `noxfile.py` and move it into `tox.ini`.

**Comment by bu3alwa** [comment]
There doesn't seem to be much to port over, looks like most of it is there. There was only one line to add. Am I missing something? Also, there is a todo deployment in  `tox.ini` is there any requirements for that?

**Comment by lzchen** [comment]
If you take a look at `noxfile.py`, the `system` and `docs` environments have not been migrated over.

As well, it would be great if we could get all or most of the CI from [opentelemetry](https://github.com/open-telemetry/opentelemetry-python/blob/master/tox.ini)
